velox/substrait/CMakeLists.txt

```diff
@@ -12,25 +12,37 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Set up Proto
-set(proto_directory ${CMAKE_CURRENT_SOURCE_DIR}/proto)
-set(substrait_proto_directory ${CMAKE_CURRENT_SOURCE_DIR}/proto/substrait)
-set(PROTO_OUTPUT_DIR "${CMAKE_CURRENT_SOURCE_DIR}/proto/")
-file(MAKE_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/proto/substrait)
-file(GLOB PROTO_FILES ${substrait_proto_directory}/*.proto
-     ${substrait_proto_directory}/extensions/*.proto)
+# Download and update substrait proto files
+FetchContent_Declare(
+  substrait
+  GIT_REPOSITORY https://github.com/substrait-io/substrait.git
+  GIT_TAG 6a74f40f0d835a447a7175b20dd430c7c39c5d11
+  SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/proto/substrait-repo
```

Comment:
You can remove `SOURCE_DIR` this will download it into the build tree but not into the source tree. 
CMake will automatically create var `substrait_SOURCE` which you can use to copy the files from. This would also make it superfluous to remove the files after copying.

Reply:
Done, Thanks